sdk/ml/azure-ai-ml/tests/component/unittests/test_data_transfer_component_entity.py

```diff
@@ -148,6 +145,7 @@ def test_copy_task_component_entity(self):
         omit_fields = [
             "properties.component_spec.$schema",
             "properties.component_spec._source",
```

Comment:
Not sure if you need to skip `$schema` and `source` in client-side hash calculation.

Reply:
I have been confirmed that if these two properties are modified, a new version of the component will be created, so it needs to be used to calculate the hash.